src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ClusterApplierServiceStatsCollector.java

```diff
@@ -0,0 +1,190 @@
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.service.ClusterApplierService;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ClusterApplierServiceStatsCollector extends PerformanceAnalyzerMetricsCollector implements
+        MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL =
+            MetricsConfiguration.CONFIG_MAP.get(ClusterApplierServiceStatsCollector.class).samplingInterval;
+    private static final Logger LOG = LogManager.getLogger(ClusterApplierServiceStatsCollector.class);
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final String GET_CLUSTER_APPLIER_SERVICE_STATS_METHOD_NAME = "getStats";
+    private final StringBuilder value;
+    private final PerformanceAnalyzerController controller;
+    private final ConfigOverridesWrapper configOverridesWrapper;
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private MetricCache metricCache;
+
+    public ClusterApplierServiceStatsCollector(PerformanceAnalyzerController controller,
+                                               ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, "ClusterApplierServiceStatsCollector");
+        value = new StringBuilder();
+        this.controller = controller;
+        this.configOverridesWrapper = configOverridesWrapper;
+        this.metricCache = new MetricCache();
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        try {
+            long mCurrT = System.currentTimeMillis();
+            if (ESResources.INSTANCE.getClusterService() == null
+                    || ESResources.INSTANCE.getClusterService().getClusterApplierService() == null) {
+                return;
+            }
+            mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            ClusterApplierServiceStats metricsObject = null;
+            try {
+                metricsObject = mapper.readValue(
+                        mapper.writeValueAsString(getClusterApplierServiceStats()), ClusterApplierServiceStats.class);
+            } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException ex) {
+                LOG.debug("No method found to get cluster state applier thread stats. " +
+                        "Skipping ClusterApplierServiceStatsCollector");
+                return;
+            }
+
+            Long latencyInMillis = computeLatency(metricsObject);
+            Long failure = computeFailure(metricsObject);
+
+            if(latencyInMillis != null && failure != null) {
```

Comment:
This will miss persisting the first metric data point as `computeLatency` and `computeFailure` will return null.